rfcs/syntax-constant-variables.md

```diff
@@ -0,0 +1,74 @@
+# Constant variables
```

Comment:
Constant variables seems like a contradictory name, how can something be constant and variable?

Reply:
I agree but it's difficult to name this something that isn't confusing; `constant variables` is wrong but at least you know what I'm talking about from a glance.

Naming it something like `constants` is a no-go because that could mean a number of things, and as far as I know there's not really any other name for these.

Reply:
Maybe do what C# does? Name it read-only variable. A `readonly` variable in C# isn't *constant*, but it's also immutable, much like Java's `final` keyword.

I think read-only meshes nicely with Roblox's existing terminology for fields and properties of their objects as well, but from the standpoint of Luau as a whole, perhaps using the existing terminology for tables and naming it "frozen variables" would be appropriate?